plugins/ecs-serviceconnect/config/netconfig.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	utils "github.com/aws/amazon-vpc-cni-plugins/plugins/utils/config"
+	cniSkel "github.com/containernetworking/cni/pkg/skel"
+	cniTypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/coreos/go-iptables/iptables"
+	"strconv"
+	"strings"
+)
+
+type NetworkMode string
+
+const (
+	// TODO: This is for experimenting with different configs. Will be changed to bridge
+	BRIDGE_DNAT   NetworkMode = "bridge_dnat"
+	BRIDGE_TPROXY NetworkMode = "bridge_tproxy"
+	AWSVPC        NetworkMode = "awsvpc"
+)
+
+// NetConfig defines the network configuration for the ecs-serviceconnect cni plugin.
+type NetConfig struct {
+	cniTypes.NetConf
+	IngressRedirectToListenerPortMap map[string]string
+	EgressPort                       int
+	EgressIPV4CIDR                   string
+	EgressIPV6CIDR                   string
+	IPProtocols                      []iptables.Protocol
+	Mode                             NetworkMode // To be used in future
```

Comment:
I'm not sure why the CNI plugin needs to know about the mode. Are we exposing the wrong abstraction? Let's talk.

Reply:
Thinking a bit more, this makes sense to me. I removed NetworkMode and will add other parameters that are applicable/needed for Bridge. Right now, modified this PR to include only awsvpc changes so that we can start testing.